core/src/hlist.rs

```diff
@@ -794,75 +805,47 @@ where
     }
 }
 
-/// An alternative to AsRef that lets us create implementations for our recursive
-/// traits that takes references, without having to deal with weird impl overflows
+/// An alternative to AsRef that does not force the reference type to be a pointer itself.
+///
+/// This lets us create implementations for our recursive traits that take the resulting
+/// Output reference type, without having to deal with strange, spurious overflows
+/// that sometimes occur when trying to implement a trait for &'a T (see this comment:
+/// https://github.com/lloydmeta/frunk/pull/106#issuecomment-377927198)
 ///
 /// This is essentially From, but the more specific nature of it means it's more ergonomic
-/// in actual usage
-pub trait ToRef {
+/// in actual usage.
```

Comment:
I might like these docs to link to the inherent method as well.  I would follow the example of `IntoReverse` (which does not describe itself as just an implementation detail), since I think it would be nice to implement this for Coproduct as well at some point later.

Reply:
That's a great idea :)